src/Indicator.vala

```diff
@@ -361,13 +422,57 @@ public class Sound.Indicator : Wingpanel.Indicator {
         }
     }
 
+    uint blubble_timeout_id = 0;
     private void play_sound_blubble () {
-        Canberra.Proplist props;
-        Canberra.Proplist.create (out props);
-        props.sets (Canberra.PROP_CANBERRA_CACHE_CONTROL, "volatile");
-        props.sets (Canberra.PROP_EVENT_ID, "audio-volume-change");
-        ca_context.play_full (0, props);
+        if (blubble_timeout_id > 0) {
+            return;
+        }
+
+        blubble_timeout_id = Timeout.add (50, () => {
+            if (!show_notification ()) { /* false when indicator open or notification cannot be shown */
+                Canberra.Proplist props;
+                Canberra.Proplist.create (out props);
+                props.sets (Canberra.PROP_CANBERRA_CACHE_CONTROL, "volatile");
+                props.sets (Canberra.PROP_EVENT_ID, "audio-volume-change");
+                ca_context.play_full (0, props);
+            }
+
+            blubble_timeout_id = 0;
+            return false;
+        });
     }
+
+    /* This also plays a sound. */
+    private bool show_notification () {
+        if (open) {
+            return false;
+        }
+
+        if (notification == null) {
+            notification = new Notify.Notification ("indicator-sound", "", "");
+            notification.set_hint ("x-canonical-private-synchronous", new Variant.string ("indicator-sound"));
+        }
+
+        if (notification != null) {
+            string icon = get_volume_icon (volume_scale.scale_widget.get_value ());
+
+            this.notification.update ("indicator-sound", "", icon);
```

Comment:
Why use `this.` here and not above?

Reply:
It is a left-over from master (current recommendation is not to use this except to disambiguate). I just forgot to delete it.